errormessages/src/ts/main.tsx

```diff
@@ -0,0 +1,112 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/// <reference path="../tsd/tsd.d.ts" />
+
+import {client, events} from 'camelot-unchained'
+import * as React from 'react';
+import * as ReactDom from 'react-dom';
+
+interface ErrorMessageProps {
+  messageList: Array<string>;
+}
+
+interface ErrorMessageState { }
+
+class ErrorMessage extends React.Component<ErrorMessageProps, ErrorMessageState> {
+
+  render() {
+       
+    var commentNodes = this.props.messageList.map(function (comment: string, index: number) {
+      return (
+        <li className='message' key={index}>{comment}</li>
+      );
+    });
+
+    return (
+      <ul id='messages'>
+        {commentNodes}
+      </ul>
```

Comment:
Made a small little test with the converted typescript: http://jsperf.com/member-function-vs-function-creation